### PR TITLE
Socket client refactor

### PR DESCRIPTION
**Pull request by NidaKarimAli**
- Updated python socket client to become AsyncClient now supporting asynchronous calls for connection and disconnection callbacks
- Updated examples to add cancel_all_orders in disconnect callback

### CODE REVIEW

File: examples/Sockets/2-orderbook_updates.py [hunk 1]
```diff
@@ -31,6 +31,8 @@ async def connection_callback():
 
     async def disconnection_callback():
         print("Sockets disconnected, performing actions...")
+        resp = resp = await client.cancel_all_orders(MARKET_SYMBOLS.ETH, [ORDER_STATUS.OPEN, ORDER_STATUS.PARTIAL_FILLED])
```
Comment (burhanuddin03) [inline examples/Sockets/2-orderbook_updates.py:34]: same as above

File: src/firefly_exchange_client/sockets.py [hunk 2]
```diff
@@ -179,9 +167,6 @@ async def unsubscribe_user_update_by_token(self, parent_account: str = None, use
                 - token: auth token generated when onboarding on firefly
         """
         try:
-            if not self.connection_established:
-                return False
-
             resp = sio.call("UNSUBSCRIBE", [
```
Comment (burhanuddin03) [inline src/firefly_exchange_client/sockets.py:170]: I think this should await as well.

File: src/firefly_exchange_client/sockets.py [hunk 3]
```diff
@@ -56,7 +56,7 @@ async def close(self):
         """
             closes the socket instance connection
         """
-        sio.disconnect()
+        await sio.disconnect()
         return
 
     @sio.on("*")
```
Comment (burhanuddin03) [inline src/firefly_exchange_client/sockets.py:62]: Can we also make this function async and call all other function with await?

**Review by burhanuddin03 — APPROVED** [review]

File: examples/Sockets/1-listening_events.py [hunk 4]
```diff
@@ -38,6 +38,8 @@ async def connection_callback():
 
     async def disconnection_callback():
         print("Sockets disconnected, performing actions...")
+        resp = await client.cancel_all_orders(MARKET_SYMBOLS.ETH, [ORDER_STATUS.OPEN, ORDER_STATUS.PARTIAL_FILLED])
```
[the remark this answers is not in the document]
Comment (MaazUddin11) [reply examples/Sockets/1-listening_events.py:41]: Ok I see you added it to a few other examples too. Just curious how'd you decide which ones needed this example?

**Review by MaazUddin11 — APPROVED** [review]

File: examples/Sockets/1-listening_events.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (NidaKarimAli) [reply examples/Sockets/1-listening_events.py:41]: Discussed on cal, only the socket examples which are affected by order placement, have placing and cancelling calls in them.

**Pull request merged**